core/src/com/unciv/logic/automation/unit/SpecificUnitAutomation.kt

```diff
@@ -144,11 +144,17 @@ object SpecificUnitAutomation {
 
         if (bestCityLocation == null) {
             // Find the best tile that is within
-            bestCityLocation = bestTilesInfo.tileRankMap.filter { bestTilesInfo.bestTile == null || it.value >= bestTilesInfo.tileRankMap[bestTilesInfo.bestTile]!! - 5 }.asSequence().sortedByDescending { it.value }.firstOrNull {
-                if (it.key in dangerousTiles && it != unit.getTile()) return@firstOrNull false
+            fun isTileRankOK(it: Map.Entry<Tile, Float>): Boolean {
+                if (it.key in dangerousTiles && it.key != unit.getTile()) return false
                 val pathSize = unit.movement.getShortestPath(it.key).size
-                return@firstOrNull pathSize in 1..3
-            }?.key
+                return pathSize in 1..3
+            }
+
+            bestCityLocation = bestTilesInfo.tileRankMap.entries.asSequence()
+                .filter { bestTilesInfo.bestTile == null || it.value >= bestTilesInfo.tileRankMap[bestTilesInfo.bestTile]!! - 5 }
```

Comment:
This line seems strange - if there is no best tile then you just take all the tiles...? What does that state actually mean...? But it's not your code so fine